### PR TITLE
Enforce NetworkX 3.4

### PR DESCRIPTION
**Pull request by f-allian**
Quick fix to #327

### CODE REVIEW

File: causal_testing/testing/causal_test_case.py [hunk 1]
```diff
@@ -47,6 +47,10 @@ def __init__(
         self.estimator = estimator
         if estimate_params is None:
             self.estimate_params = {}
+
+        else:
+            self.estimate_params = estimate_params
```
Comment (jmafoster1) [inline causal_testing/testing/causal_test_case.py:52]: @f-allian please could you add a couple of tests for this? Maybe one explicitly setting it to empty and one setting a parameter for something (e.g. "formula" for the `LinearRegressionEstimator` class?

Comment (f-allian) [reply causal_testing/testing/causal_test_case.py:52]: @jmafoster1 Good spot - sure.

**Review by jmafoster1 — APPROVED** [review]
I modified the assertions a bit on the tests so that the test case is actually run. I realised that the `formula` parameter wasn't actually a valid estimate parameter for `estimate_ate`, so wanted to test the parameters were actually getting used properly.
